test/pylib/coverage_utils.py

```diff
@@ -0,0 +1,1293 @@
+
+        Args:
+            f (PathLike): The path to the file to be classified
+
+        Returns:
+            FileType: One of filetype, if the file couldn't be recognized,
+            FileType.UNRECOGNIZED will be returned.
+        """
+        file_description = await FileType.get_file_description(f)
+        for ft in FileType:
+            if m := ft.value[0].fullmatch(file_description):
+                if ft in ELF_TYPES:
+                    profiled =  "PROFILED_" if await FileType.is_profiled(f) else ""
+                    debug_only = "DEBUG_ONLY_" if await FileType.is_debug_only(f) else ""
+                    return FileType[debug_only + profiled + ft.name]
+                return ft
+        return FileType.UNRECOGNIZED
+
+
+ELF_TYPES = [FileType.EXEC_BIN,FileType.EXEC_SO]
+PROFILED_ELF_TYPES = [FileType.PROFILED_EXEC_BIN, FileType.PROFILED_EXEC_SO]
+
+# Convenience functions
+
+def convert_to_path_list(paths : PathLikeOrPathLikeCollection) -> List[Path]:
+    """ A convenience function to normalize PathLikeOrPathLikeCollection into
+        a PathCollection
+
+    Args:
+        paths (PathLikeOrPathLikeCollection): The parameter to convert
+
+    Returns:
+        List[Path]: A path list
+    """
+    if paths is None:
+        return []
+    elif isinstance(paths, str) or not isinstance(paths, IterableType):
+        return [Path(paths)]
+    return [ Path(path) for path in paths]
+
+def concurrency_to_semaphore(concurrency : ConcurrencyParam) -> asyncio.Semaphore:
+    """A convenience function to convert a `ConcurrencyParam` to be used to limit
+    concurrency of execution.
+
+    Args:
+        concurrency (ConcurrencyParam): The concurrency to be converted to a semaphore.
+
+    Returns:
+        asyncio.Semaphore: A semaphore according to the following semantics:
+        1. If concurrence's actual type is int, so a semaphore with `concurrency` units.
+        2. If `concurrency` is a semaphore, it will be returned as is
+        3. In any other case, a semaphore with 1 unit will be returned (no concurrency allow)
+    """
+    if isinstance(concurrency,int):
+        return asyncio.Semaphore(concurrency)
+    elif isinstance(concurrency,asyncio.Semaphore):
+        return concurrency
+    else:
+        return asyncio.Semaphore(1)
+
+async def verify_file_type(f : PathLike, file_type: Union[FileType,Iterable[FileType]]) -> bool:
+    """Verify that a file is of certain type
+
+    Args:
+        f (PathLike): The path to the file whose type needs to be verified
+        file_type (Union[FileType,Iterable[FileType]]): A collection of filetypes or a single
+        file type for the file to be verified against.
+
+    Returns:
+        bool: True if the file type matches one of `file_type` False otherwise.
+    """
+    if not isinstance(file_type,IterableType):
+        file_type = [ file_type ]
+    file_type = list(file_type)
+    return await FileType.get_file_type(f) in file_type
```

Comment:
The reason this helper exists is because there is no "async lambda", Since get_file_type uses system process under the hood. The idea is to parallelize multiple "verifies".
If this function doesn't exists, lines like this will not be possible:
```
 profiles_filter = await gather_limited_concurrency(
        *(
            verify_file_type(f, [FileType.RAW_PROFILE, FileType.INDEXED_PROFILE])
            for f in profiles
        ),
        concurrency = sem,
    )
```
However, I do see a pattern of the usage of this function, I always, paralelly verify file types and then
use the other function you didn't like (:slightly_smiling_face: ), `filter_from_list` to get only the files I need.
So actually I can probably can unify them to one function with `verify_file_type` being an internal function in this.
I think this is what I'll do.

Reply:
Removed all of those functions as I found out they are used only in very specific places.